src/Microsoft.Diagnostics.Monitoring.Options/MonitorCapability.cs

```diff
@@ -1,18 +1,22 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.ComponentModel.DataAnnotations;
 using System.Text.Json.Serialization;
 
 namespace Microsoft.Diagnostics.Monitoring.Options
 {
-    public class MonitorCapability
+    public class MonitorCapability : IMonitorCapability
     {
-        [Required]
         [JsonPropertyName("name")]
-        public string Name { get; set; } = string.Empty;
+        public string Name { get; init; }
```

Comment:
I would add the `[Required]` attribute back to make the OpenAPI description such that the property is not nullable.